tests/test_onion_message.py

```diff
@@ -0,0 +1,255 @@
+                                                                              with_blinded_node_ids=True)
+
+        self.assertEqual(hop_shared_secrets, [
+            bfh('c04d2a4c518241cb49f2800eea92554cb543f268b4c73f85693541e86d649205'),
+            bfh('196f1f3e0be9d65f88463c1ab63e07f41b4e7c0368c28c3e6aa290cc0d22eaed'),
+            bfh('c7b33d74a723e26331a91c15ae5bc77db28a18b801b6bc5cd5bba98418303a9d'),
+            bfh('024955ed0d4ebbfab13498f5d7aacd00bf096c8d9ed0473cdfc96d90053c86b7')
+        ])
+
+        self.assertEqual(blinded_node_ids, [
+            bfh('02d1c3d73f8cac67e7c5b6ec517282d5ba0a52b06a29ec92ff01e12decf76003c1'),
+            bfh('03f1465ca5cf3ec83f16f9343d02e6c24b76993a93e1dea2398f3147a9be893d7a'),
+            bfh('035dbc0493aa4e7eea369d6a06e8013fd03e66a5eea91c455ed65950c4942b624b'),
+            bfh('0237bf019fa0fbecde8b4a1c7b197c9c1c76f9a23d67dd55bb5e42e1f50bb771a6')
+        ])
+
+        hops_data = [
+            OnionHopsDataSingle(tlv_stream_name='onionmsg_tlv',
+                                blind_fields={'next_node_id': {'node_id': bfh('0324653eac434488002cc06bbfb7f10fe18991e35f9fe4302dbea6d2353dc0ab1c')},
+                                              'next_blinding_override': {'blinding': bfh('031b84c5567b126440995d3ed5aaba0565d71e1834604819ff9c17f5e9d5dd078f')},
+                                              }
+                                ),
+            OnionHopsDataSingle(tlv_stream_name='onionmsg_tlv',
+                                blind_fields={'next_node_id': {'node_id': bfh('027f31ebc5462c1fdce1b737ecff52d37d75dea43ce11c74d25aa297165faa2007')},
+                                              'unknown_tag_561': {'data': bfh('123456')}
+                                              }
+                                ),
+            OnionHopsDataSingle(tlv_stream_name='onionmsg_tlv',
+                                blind_fields={'padding': {'padding': bfh('0000000000')},
+                                              'next_node_id': {'node_id': bfh('032c0b7cf95324a07d05398b240174dc0c2be444d96b159aa6c7f7b1e668680991')}
+                                              }
+                                ),
+            OnionHopsDataSingle(tlv_stream_name='onionmsg_tlv', payload={'message': {'text': b'hello'}},
+                                blind_fields={'padding': {'padding': b''},
+                                              'path_id': {'data': bfh('deadbeefbadc0ffeedeadbeefbadc0ffeedeadbeefbadc0ffeedeadbeefbadc0')},
+                                              'unknown_tag_65535': {'data': bfh('06c1')}
+                                              }
+                                )
+        ]
+
+        hop_shared_secrets, blinded_node_ids = get_shared_secrets_along_route(payment_path_pubkeys, session_key,
+                                                                              with_blinded_node_ids=True)
```

Comment:
why is this called twice? this looks unnecessary

Reply:
indeed, not necessary